packages/tom-server/src/user-info-api/controllers/index.ts

```diff
@@ -31,19 +44,56 @@ class UserInfoController implements IUserInfoController {
   ): Promise<void> => {
     try {
       const { userId } = req.params
-      if (userId !== req.userId) {
+      if (req.userId == null) {
         res.status(403).json({ error: errCodes.forbidden })
```

Comment:
I'd advocate for a `400 M_MISSING_PARAMS` Matrix error instead
```suggestion
        res.status(400).json({ error: errCodes.missingParams })
```